cnd/src/swap_protocols/rfc003/create_swap.rs

```diff
@@ -251,20 +251,45 @@ where
     pub alpha_asset: AA,
     pub beta_asset: BA,
     pub hash_function: HashFunction,
-    pub alpha_ledger_redeem_identity: AL::Identity,
-    pub alpha_ledger_refund_identity: AL::Identity,
-    pub beta_ledger_redeem_identity: BL::Identity,
-    pub beta_ledger_refund_identity: BL::Identity,
+    pub alpha_ledger_redeem_identity: AI,
+    pub alpha_ledger_refund_identity: AI,
+    pub beta_ledger_redeem_identity: BI,
+    pub beta_ledger_refund_identity: BI,
     pub alpha_expiry: Timestamp,
     pub beta_expiry: Timestamp,
     pub secret_hash: SecretHash,
 }
 
-impl<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset> OngoingSwap<AL, BL, AA, BA> {
-    pub fn new(
-        request: Request<AL, BL, AA, BA>,
-        accept: Accept<AL::Identity, BL::Identity>,
-    ) -> Self {
+impl<AL, BL, AI, BI, AA, BA> OngoingSwap<AL, BL, AI, BI, AA, BA>
+where
+    AL: Ledger<Identity = AI>,
+    BL: Ledger<Identity = BI>,
+    AI: Clone
```

Comment:
Ouch! :D

Should we stop here and do https://github.com/comit-network/comit-rs/issues/2065 first? That would allow you to omit these trait bounds here ...

Reply:
If you think #2065 helps (I haven't checked) then I agree to stop here.